File: nba_statmaker.py
from flask import Flask, jsonify, request 
from flask_cors import CORS
import requests

app = Flask(__name__)
CORS(app)  

class NBAStatMaker:
    def __init__(self, auth_key):
        self.auth_key = auth_key
        self.base_url = "https://api.balldontlie.io/v1/players"

    """
    Once the user enters a player's name, we return all the players in the api that match or have partial 
    names to the one entered by the user.
    """
    def get_player_id(self, search_term):
        url = f"{self.base_url}?search={search_term}"
        headers = {"Authorization": self.auth_key}  
        response = requests.get(url, headers=headers)  

        if response.status_code == 200:
            data = response.json()
            return data['data']  
        else:
            return None 

    """
    After the specific player is selected, return the information of the player.
    """
    def get_player_info(self, player_id):
        url = f"{self.base_url}/{player_id}"
        headers = {"Authorization": self.auth_key} 
        response = requests.get(url, headers=headers)  

        if response.status_code == 200:
            data = response.json()
            return data['data']  
        else:
            return None  # Handle failure

# Initialize NBAStatMaker
nba_stat_maker = NBAStatMaker(auth_key="b2b9782d-fa82-470f-9092-411c9a66a986")

"""
Flask route to search for the player 
"""
@app.route('/search_player', methods=['GET'])
def search_player():
    search_term = request.args.get('search')
    players = nba_stat_maker.get_player_id(search_term)
    if players:
        return jsonify({'data': players}), 200
    else:
        return jsonify({"error": "No players found"}), 404

"""
Flask route to return the player's information
"""
@app.route('/get_player_info/<int:player_id>', methods=['GET'])
def get_player_info(player_id):
    player_info = nba_stat_maker.get_player_info(player_id)
    if player_info:
        return jsonify(player_info), 200
    else:
        return jsonify({"error": "Player not found"}), 404

if __name__ == '__main__':
    app.run(debug=True)

File: nba_statmaker.js
const API_KEY = "b2b9782d-fa82-470f-9092-411c9a66a986"; 

document.getElementById("player-form").addEventListener("submit", function(event) {
    event.preventDefault(); 

    let playerName = document.getElementById("playername").value;
    if (playerName === "") {
        alert("Please enter a player's name");
        return;
    }

    // Clear previous search results
    document.getElementById("players-table").innerHTML = `
        <tr>
            <th>Player Name</th>
            <th>Position</th>
            <th>Select</th>
        </tr>
    `;

    // Get players matching to player name requested by user and display
    fetch(`http://ec2-3-84-6-10.compute-1.amazonaws.com/search_player?search=${playerName}`)
    .then(response => response.json())
    .then(data => {
        if (data.data && data.data.length > 0) {
            data.data.forEach((player) => {
                
                let playerRow = `
                    <tr>
                        <td>${player.first_name} ${player.last_name}</td>
                        <td>${player.position ? player.position : 'N/A'}</td>
                        <td><button onclick="getPlayerInfo(${player.id})">Select</button></td>
                    </tr>
                `;
                document.getElementById("players-table").innerHTML += playerRow;
            });
        } else {
            alert("No matching players found");
        }
    })
    .catch(error => {
        console.error("Error fetching player data:", error);
        alert("An error occurred while fetching player data.");
    });
});

// Once specific player is selected, populate table with that player's information.
function getPlayerInfo(playerId) {
    fetch(`http://ec2-3-84-6-10.compute-1.amazonaws.com/get_player_info/${playerId}`)
    .then(response => {
        if (!response.ok) {
            throw new Error('Network response was not ok');
        }
        return response.json();
    })
    .then(player => {
        // Clear the player info table and insert new details
        document.getElementById("player-info-table").innerHTML = `
            <tr>
                <th>Player Name</th>
                <th>Position</th>
                <th>Height</th>
                <th>Weight</th>
                <th>Team</th>
                <th>Jersey Number</th>
                <th>College</th>
                <th>Draft Year</th>
                <th>Draft Round</th>
                <th>Draft Pick</th>
            </tr>
            <tr>
                <td>${player.first_name} ${player.last_name}</td>
                <td>${player.position ? player.position : 'N/A'}</td>
                <td>${player.height ? player.height : 'N/A'}</td>
                <td>${player.weight ? player.weight + ' lbs' : 'N/A'}</td>
                <td>${player.team ? player.team.full_name : 'N/A'}</td>
                <td>${player.jersey_number ? player.jersey_number : 'N/A'}</td>
                <td>${player.college ? player.college : 'N/A'}</td>
                <td>${player.draft_year ? player.draft_year : 'N/A'}</td>
                <td>${player.draft_round ? player.draft_round : 'N/A'}</td>
                <td>${player.draft_number ? player.draft_number : 'N/A'}</td>
            </tr>
        `;
    })
    .catch(error => {
        console.error("Error fetching detailed player info:", error);
        alert("An error occurred while fetching detailed player info.");
    });
}